utils/src/main/java/software/amazon/awssdk/utils/StringUtils.java

```diff
@@ -603,4 +620,142 @@ public static String uncapitalize(final String str) {
         }
         return new String(newCodePoints, 0, outOffset);
     }
+
+    /**
+     * This method appends a string to a string builder and collapses contiguous
+     * white space is a single space.
+     *
+     * This is equivalent to:
+     *      destination.append(source.replaceAll("\\s+", " "))
+     * but does not create a Pattern object that needs to compile the match
+     * string; it also prevents us from having to make a Matcher object as well.
+     *
+     */
+    public static void appendCompactedString(final StringBuilder destination, final String source) {
```

Comment:
This is only used in the signer right now, and I'm not sure where else we'd use it. Should we migrate it to there for now?